staging/src/k8s.io/sample-controller/artifacts/examples/crd.yaml

```diff
@@ -0,0 +1,11 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: nginxs.example.controller.code-generator.k8s.io
+spec:
+  group: example.controller.code-generator.k8s.io
+  version: v1alpha1
+  names:
+    kind: NGINX
```

Comment:
nit: not sure about any best practices but can we have this as `nginx` instead of `NGINX`? That way the plural (`nginxs`) seems good. Else, we should probably make it `NGINXS`.

Reply:
Or should it be `nginxes`? :)

Reply:
I'm not sure on the best thing to do here - nginxs works, I suppose, but does sound odd. I've never seen nginxes before. I wonder what the 'official' plural of NGINX is 😆 

I did try changing the `kind` field in the CRD to `nginx`, but upon running the controller get:

```k8s.io/kubernetes/vendor/k8s.io/sample-controller/pkg/client/informers/externalversions/factory.go:61: Failed to list *v1alpha1.NGINX: no kind "nginxList" is registered for version "example.controller.code-generator.k8s.io/v1alpha1"```

I've not had much time to dig into this right now, perhaps it's down to me still using the struct name NGINX and NGINXList in types.go?

I'm half thinking we should rename our resource to FooBar or something, although I'm not too sure...

Reply:
> perhaps it's down to me still using the struct name NGINX and NGINXList in types.go?

Yup, should be just this.

> I'm half thinking we should rename our resource to FooBar or something, although I'm not too sure...

Yeah, I suppose a rename would be nice. :)